frame/elections-phragmen/src/lib.rs

```diff
@@ -1069,25 +1103,29 @@ impl<T: Config> Pallet<T> {
 					// fetch deposits from the one recorded one. This will make sure that a
 					// candidate who submitted candidacy before a change to candidacy deposit will
 					// have the correct amount recorded.
-					<Members<T>>::put(
+					<Members<T>>::put::<BoundedVec<_, T::DesiredMembers>>(
 						new_members_sorted_by_id
 							.iter()
 							.map(|(who, stake)| SeatHolder {
 								deposit: deposit_of_candidate(who),
 								who: who.clone(),
 								stake: *stake,
 							})
-							.collect::<Vec<_>>(),
+							.collect::<Vec<_>>()
+							.try_into()
+							.expect("Cannot accept more than DesiredMembers."),
 					);
-					<RunnersUp<T>>::put(
+					<RunnersUp<T>>::put::<BoundedVec<_, T::DesiredRunnersUp>>(
 						new_runners_up_sorted_by_rank
 							.into_iter()
 							.map(|(who, stake)| SeatHolder {
 								deposit: deposit_of_candidate(&who),
 								who,
 								stake,
 							})
-							.collect::<Vec<_>>(),
+							.collect::<Vec<_>>()
+							.try_into()
+							.expect("Cannot accept more than DesiredRunnersUp."),
```

Comment:
Same as above, we should not panic here.

Reply:
I think that we can use expect here since the error is handled later in the code.

https://github.com/paritytech/substrate/blob/5683513310f44eb2dc8026cd79b189b196a5982a/frame/elections-phragmen/src/lib.rs#L1099-L1106

Reply:
An `except` is not an error, it panics.  
See [[When to use .expect()](https://substrate.stackexchange.com/questions/317/when-to-use-expect)](https://substrate.stackexchange.com/questions/317/when-to-use-expect).